File: profiler/manage_user_sets.py
# With this sample, you can manage user sets, such as obtaining the names of all user sets, adding a user set, switching the user set, and saving parameter settings to the user set.

from mecheye.shared import *
from mecheye.profiler import *
from mecheye.profiler_utils import *


class ManageUserSets(object):
    def __init__(self):
        self.profiler = Profiler()

    def main(self):
        if not find_and_connect(self.profiler):
            return

        user_set_manager = self.profiler.user_set_manager()
        error, user_sets = user_set_manager.get_all_user_set_names()
        show_error(error)

        print("All user sets: ")
        for user_set in user_sets:
            print(user_set)

        current_user_set = self.profiler.current_user_set()
        error, name = current_user_set.get_name()
        print("Current user set: ", name)

        success_message = "Set \\" + user_sets[0] +"\\ as the current user set."
        show_error(user_set_manager.select_user_set(user_sets[0]), success_message)

        success_message = "Save all parameters to current user set."
        show_error(current_user_set.save_all_parameters_to_device(), success_message)

        self.profiler.disconnect()
        print("Disconnected form the Mech-Eye Profiler successfully")

if __name__ == '__main__':
    a = ManageUserSets()
    a.main()

File: profiler/ManageUserSets/ManageUserSets.cpp
/*******************************************************************************
 *BSD 3-Clause License
 *
 *Copyright (c) 2016-2024, Mech-Mind Robotics
 *All rights reserved.
 *
 *Redistribution and use in source and binary forms, with or without
 *modification, are permitted provided that the following conditions are met:
 *
 *1. Redistributions of source code must retain the above copyright notice, this
 *   list of conditions and the following disclaimer.
 *
 *2. Redistributions in binary form must reproduce the above copyright notice,
 *   this list of conditions and the following disclaimer in the documentation
 *   and/or other materials provided with the distribution.
 *
 *3. Neither the name of the copyright holder nor the names of its
 *   contributors may be used to endorse or promote products derived from
 *   this software without specific prior written permission.
 *
 *THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
 *AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
 *IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
 *DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
 *FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
 *DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
 *SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
 *CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
 *OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
 *OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 ******************************************************************************/

/*
With this sample, you can manage user sets, such as obtaining the names of all parameter
groups, adding a user set, switching the user set, and saving parameter settings to
the user set.
*/

#include "profiler/Profiler.h"
#include "profiler/api_util.h"
#include <iostream>

int main()
{
    mmind::eye::Profiler profiler;
    if (!findAndConnect(profiler))
        return -1;

    mmind::eye::UserSetManager& userSetManager = profiler.userSetManager();

    // Obtain the names of all user sets.
    std::vector<std::string> userSets;
    showError(userSetManager.getAllUserSetNames(userSets));

    std::cout << "All user sets: ";
    for (size_t i = 0; i < userSets.size(); i++)
        std::cout << userSets[i] << "  ";
    std::cout << std::endl;

    mmind::eye::UserSet& curSettings = userSetManager.currentUserSet();

    // Obtain the name of the currently selected user set.
    std::string currentName;
    showError(curSettings.getName(currentName));
    std::cout << "The current user set: " << currentName << std::endl;

    // Add a user set.
    const std::string newSetting{"NewSettings"};
    showError(userSetManager.addUserSet(newSetting));
    std::cout << "Add a new user set with the name of \"" << newSetting << "\"." << std::endl;

    // Select a user set by its name.
    showError(userSetManager.selectUserSet(newSetting));
    std::cout << "Select the \"" << newSetting << "\" user set." << std::endl;

    showError(curSettings.saveAllParametersToDevice());
    std::cout << "Save all parameters to the current user set." << std::endl;

    profiler.disconnect();
    std::cout << "Disconnected from the laser profiler successfully." << std::endl;
    return 0;
}
